info/src/main/java/edu/artic/info/AccessMemberCardFragment.kt

```diff
@@ -0,0 +1,203 @@
+import com.fuzz.rx.defaultThrottle
+import com.fuzz.rx.disposedBy
+import com.google.zxing.BarcodeFormat
+import com.jakewharton.rxbinding2.view.clicks
+import com.jakewharton.rxbinding2.widget.text
+import com.jakewharton.rxbinding2.widget.textChanges
+import com.journeyapps.barcodescanner.BarcodeEncoder
+import edu.artic.analytics.ScreenCategoryName
+import edu.artic.base.utils.hideSoftKeyboard
+import edu.artic.viewmodel.BaseViewModelFragment
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.rxkotlin.subscribeBy
+import kotlinx.android.synthetic.main.fragment_validate_member_information.*
+import kotlinx.android.synthetic.main.layout_member_access_card.*
+import kotlinx.android.synthetic.main.layout_member_information_form.*
+import timber.log.Timber
+import kotlin.reflect.KClass
+
+
+/**
+ * This fragment is responsible for:
+ *
+ * 1. Displays the member information form (with memberID & ZIP code fields) for validation .
+ * 2. Displays the cardHolder information with the barcode (encodes memberID to PDF417) and hides the form.
+ */
+class AccessMemberCardFragment : BaseViewModelFragment<AccessMemberCardViewModel>() {
+    override val viewModelClass: KClass<AccessMemberCardViewModel> = AccessMemberCardViewModel::class
+    override val title: String = "Access Member Card"
+    override val layoutResId: Int = R.layout.fragment_validate_member_information
+    override val screenCategory: ScreenCategoryName? = null
+
+    private lateinit var toolbarColorAnimator: ValueAnimator
+
+    override fun setupBindings(viewModel: AccessMemberCardViewModel) {
+        super.setupBindings(viewModel)
+
+        viewModel.isValid
+                .subscribeBy {
+                    signIn.isEnabled = it
+                }.disposedBy(disposeBag)
+
+        zipCode.textChanges()
+                .skipInitialValue()
+                .map { it.toString() }
+                .bindTo(viewModel.zipCode)
+                .disposedBy(disposeBag)
+
+        memberId.textChanges()
+                .skipInitialValue()
+                .map { it.toString() }
+                .bindTo(viewModel.memberID)
+                .disposedBy(disposeBag)
+
+        viewModel.loadStatus
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeBy { loadStatus ->
+                    when (loadStatus) {
+                        is LoadStatus.Loading -> {
+                            loading.visibility = View.VISIBLE
+                            mask.visibility = View.VISIBLE
+                        }
+                        is LoadStatus.Error -> {
+                            loading.visibility = View.GONE
+                            mask.visibility = View.GONE
+                            /**
+                             * Designs does not exist for error dialogs yet.
+                             * Temporarily using [AlertDialog].
+                             * TODO:: Updated dialog once design is finalized.
```

Comment:
Right. We'll need it to be a DialogFragment too, then.

Reply:
Unsure if this is change request or comment or just note for yourself @Cliabhach, please be more specific or use some kind of prefix to a comment like @agrosner 
pp for personal preference
? for question
r - required change

Reply:
It's a comment since I'm replying to the `TODO`. Github has its own way to leave non-binding comments but it sure doesn't make that easy...